cardano-testnet/src/Testnet/Components/Configuration.hs

```diff
@@ -149,27 +145,26 @@ createSPOGenesisAndFiles (NumPools numPoolNodes) era shelleyGenesis alonzoGenesi
     , "--out-dir", tempAbsPath
     ]
 
-  -- Here we move all of the keys etc generated by create-testnet-data
-  -- for the nodes to use
+  -- Remove the input files. We don't need them anymore, since create-testnet-data wrote new versions.
+  forM_ [inputGenesisShelleyFp, inputGenesisAlonzoFp, inputGenesisConwayFp] (liftIO . System.removeFile)
 
   -- Move all genesis related files
-
   genesisByronDir <- H.createDirectoryIfMissing $ tempAbsPath </> "byron"
 
   files <- H.listDirectory tempAbsPath
-  forM_ files $ \file -> do
-    H.note file
+  forM_ files H.note
 
   H.renameFile (tempAbsPath </> "byron-gen-command" </> "genesis.json") (genesisByronDir </> "genesis.json")
-  H.renameFile (tempAbsPath </> "shelley-genesis.json") (genesisShelleyDir </> "genesis.shelley.json")
 
   -- For some reason when setting "--total-supply 10E16" in create-testnet-data, we're getting negative
-  -- treasury. TODO: This should be fixed by https://github.com/IntersectMBO/cardano-cli/pull/644
-  -- So this can be removed when cardano-cli is upgraded above 8.20.3.0.
-  H.rewriteJsonFile @Value (genesisShelleyDir </> "genesis.shelley.json") $ \o -> o
+  -- treasury. TODO: This was supposed to be fixed by https://github.com/IntersectMBO/cardano-cli/pull/644,
+  -- but no, it's still there.
+  H.rewriteJsonFile @Value (tempAbsPath </> defaultGenesisFilepath ShelleyEra) $ \o -> o
     & L.key "maxLovelaceSupply" . L._Integer .~ 10_000_000_000_000_000
```

Comment:
I tried to get rid of this rewrite, but it makes the `Treasury Growth` fail so something is still wrong somewhere. I will look into it in another PR. I don't want to block this one.